src/contracts/Security/AuthorizationHeaderRESTRequestMatcher.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Contracts\Rest\Security;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestMatcher;
+
+final class AuthorizationHeaderRESTRequestMatcher extends RequestMatcher
+{
+    public function matches(Request $request): bool
+    {
+        if ($request->attributes->get('is_rest_request', false) !== true) {
+            return false;
+        }
+
+        if ($request->attributes->get('_route') === 'ibexa.rest.create_token'
+            || $request->headers->has('Authorization')) {
```

Comment:
Subjectively I feel this way of writing long if conditions is more readable to me:
```suggestion
        if (
            $request->attributes->get('_route') === 'ibexa.rest.create_token' ||
            $request->headers->has('Authorization')
        ) {
```

Reply:
Personally I think that logic operators should come first, as they are important for the reader to indicate how statements are connected.

Additionally, it is easier for git to understand the changes, as you would be changing only one line - not two - when removing a condition from the list.

Reply:
Changed slightly to match PSR-12 in b8ff4bea.